src/config.ts

```diff
@@ -0,0 +1,101 @@
+import mergeOptions from 'merge-options'
+import { dnsaddrResolver } from '@multiformats/multiaddr/resolvers'
+import * as Constants from './constants.js'
+import { AGENT_VERSION } from './identify/consts.js'
+import * as RelayConstants from './circuit/constants.js'
+import { publicAddressesFirst } from '@libp2p/utils/address-sort'
+import { FAULT_TOLERANCE } from './transport-manager.js'
+import type { Multiaddr } from '@multiformats/multiaddr'
+import type { Libp2pInit } from './index.js'
+import { codes, messages } from './errors.js'
+import errCode from 'err-code'
+import type { RecursivePartial } from '@libp2p/interfaces'
+
+const DefaultConfig: Partial<Libp2pInit> = {
```

Comment:
How come you put Partial here? 

Shouldn't we enforce that whoever adds a new module must put defaults here? Maybe even create some type which would make everything mandatory here 😄

Reply:
Without `Partial` is becomes necessary to define things things like `.peerId` and `.datastore` that aren't simple values.

Reply:
```suggestion
const DefaultConfig: Required<Omit<Libp2pInit, "peer" | "datastore">> & Libp2pInit = {
```
would probably do the trick but I don't have a strong preference, it might end up being too hard to maintain what to keep optional.

Reply:
Yeah, it starts getting a bit fiddly like this, maybe let's look at it again later.